src/main/java/org/jenkinsci/plugins/nomad/NomadCloud.java

```diff
@@ -335,7 +399,8 @@ public Node call() throws Exception {
 
             // Support for Jenkins security
             String jnlpSecret = "";
-            if (Jenkins.get().isUseSecurity()) {
+            if (Jenkins.get().isUseSecurity() || template.isDockerDriver()) {
```

Comment:
I'm not sure about this condition at all. It depends very much on how the agent.jar is executed. When java -cp is used then the secret is mandatory but when -jar is used then it's optional I guess. It probably makes sense to refactor this part in the future so that either the cp or the jar approach is used but not both.